molgenis-swagger/src/main/java/org/molgenis/swagger/controller/SwaggerController.java

```diff
@@ -0,0 +1,80 @@
+package org.molgenis.swagger.controller;
+
+import org.molgenis.data.meta.MetaDataService;
+import org.molgenis.data.meta.model.EntityType;
+import org.molgenis.security.core.token.TokenService;
+import org.molgenis.security.core.utils.SecurityUtils;
+import org.molgenis.ui.MolgenisPluginController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.swagger.controller.SwaggerController.URI;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+
+/**
+ * Serves Swagger documentation of the REST API.
+ * @see  <a href="http://swagger.io/">http://swagger.io/</a>
+ */
+@Controller
+@RequestMapping(URI)
+public class SwaggerController extends MolgenisPluginController
+{
+
+	private static final String ID = "swagger";
+	public static final String URI = MolgenisPluginController.PLUGIN_URI_PREFIX + ID;
+
+	private final MetaDataService metaDataService;
+	private final TokenService tokenService;
+
+	@Autowired
+	public SwaggerController(MetaDataService metaDataService, TokenService tokenService)
+	{
+		super(URI);
+		this.metaDataService = requireNonNull(metaDataService);
+		this.tokenService = requireNonNull(tokenService);
+	}
+
+	/**
+	 * Serves the Swagger UI which allows you to try out the documented endpoints.
+	 * Sets the url parameter to the swagger yaml that describes the REST API.
+	 * Creates an apiKey token for the current user.
+	 */
+	@RequestMapping
+	public String init(Model model)
+	{
+		model.addAttribute("url",
```

Comment:
Next line is very long, my formatter thinks it should look like this. Yours disagrees?